### PR TITLE
Template tilt cover

### LINKED ISSUE #42: Template cover SET_COVER_POSITION support
Do you have any plans to support [setting cover position](https://www.home-assistant.io/components/cover/#service-coverset_cover_position) for template covers? Additionally, is there a good way to tie a cover position sensor to the position of a slider in the Home Assistant UI? It seems like existing options require the sensor/slider to be pretty much separate from the cover and its open/closed state and open/stop/close buttons.

### PR DESCRIPTION
**Pull request by mtl010957**
## Description:
I think this is the simplest way to enable tilt control in the existing [Template Cover](https://esphome.io/components/cover/template.html) platform. This adds two new configuration options, ```tilt_action``` and ```tilt_lambda```. ```tilt_action``` expects some action to respond to a tilt position command, usually would be setting an output. ```tilt_lambda``` holds code to compute the actual tilt position to send back to the frontend.

**Related issue (if applicable):** fixes [feature-request #42](https://github.com/esphome/feature-requests/issues/42)

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#260

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/template/cover/__init__.py [hunk 1]
```diff
@@ -63,6 +73,9 @@ def cover_template_publish_to_code(config, action_id, template_arg, args):
     if CONF_POSITION in config:
         template_ = yield cg.templatable(config[CONF_POSITION], args, float)
         cg.add(var.set_position(template_))
+    if CONF_TILT in config:
```
Comment (OttoWinter) [inline esphome/components/template/cover/__init__.py]: CONF_TILT also needs to be added to the schma above (line 65)

**Review by OttoWinter — COMMENTED** [review]
@mtl010957 Oops, sorry. I thought this PR was a WIP because the changes in .cpp were missing - but as it turns out they are already there 😅

**Comment by mtl010957** [comment]
@OttoWinter  Hey, no worries, I saw you had quite a few other big changes in the works so I was in no rush. I'm traveling for a few days so I'll clean this up and ping when I get back.  Thanks

File: esphome/components/template/cover/__init__.py (hunk 1 above)
Comment (mtl010957) [reply esphome/components/template/cover/__init__.py]: @OttoWinter  I've added the CONF_TILT to the schema, this is ready for review now.  Thanks.

**Pull request merged**